### PR TITLE
@W-8122073@ Fix csv output escaping issue

### PR DESCRIPTION
**Pull request by jbartolotta-sfdc**
Fix issues with csv output. Commas were converted to semicolons and quotes weren't escaped.

Added additional escaping for XML output.

Added more tests.

### CODE REVIEW

File: src/lib/RuleResultRecombinator.ts [hunk 1]
```diff
@@ -78,16 +79,19 @@ export class RuleResultRecombinator {
 		for (const result of results) {
 			const from = process.cwd();
 			const fileName = path.relative(from, result.fileName);
+			const escapedFileName = this.safeHtmlEscape(fileName);
```
Comment (jbartolotta-sfdc) [inline src/lib/RuleResultRecombinator.ts:82]: We should consider importing an xml package that will do this for us. Added extra escaping found in tests as a short term fix.

File: src/lib/RuleResultRecombinator.ts [hunk 2]
```diff
@@ -78,16 +79,19 @@ export class RuleResultRecombinator {
...
 			for (const v of result.violations) {
+				const escapedMessage = this.safeHtmlEscape(v.message.trim());
+				const escapedCategory = this.safeHtmlEscape(v.category);
+				const escapedRuleName = this.safeHtmlEscape(v.ruleName);
+				const escapedUrl = this.safeHtmlEscape(v.url);
+
 				problemCount++;
-				violations += `
-            <violation severity="${v.severity}" line="${v.line}" column="${v.column}" endLine="${v.endLine}" endColumn="${v.endColumn}" rule="${v.ruleName}" category="${v.category}" url="${v.url}">
-${v.message.trim()}
```
Comment (jbartolotta-sfdc) [inline src/lib/RuleResultRecombinator.ts:86]: This was creating extra newlines in the message.

File: src/lib/RuleResultRecombinator.ts [hunk 3]
```diff
@@ -78,16 +79,19 @@ export class RuleResultRecombinator {
...
-            </violation>`;
+				violations += `<violation severity="${v.severity}" line="${v.line}" column="${v.column}" endLine="${v.endLine}" endColumn="${v.endColumn}" rule="${escapedRuleName}" category="${escapedCategory}" url="${escapedUrl}">${escapedMessage}</violation>`;
 			}
 			resultXml += `
-      <result file="${fileName}">
+      <result file="${escapedFileName}" engine="${result.engine}">
```
Comment (jbartolotta-sfdc) [inline src/lib/RuleResultRecombinator.ts:94]: Added missing message.

File: src/lib/RuleResultRecombinator.ts [hunk 4]
```diff
@@ -248,27 +252,38 @@ URL: ${url}
 		return Mustache.render(template, templateData);
 	}
 
-	private static constructCsv(results: RuleResult[]): string {
+	private static async constructCsv(results: RuleResult[]): Promise<string> {
```
Comment (jbartolotta-sfdc) [inline src/lib/RuleResultRecombinator.ts:255]: This is the bug fix. Use csv-stringify to create the output.

**Review by jfeingold35 — APPROVED** [review]

**Pull request merged**